File: src/main/java/org/example/FixedLengthParser.java
package org.example;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class FixedLengthParser {
    public List<Record> parseFile(String filePath) throws IOException {
        List<Record> records = new ArrayList<>();
        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
            String line;
            while ((line = reader.readLine()) != null) {
                if (line.length() != 25) {
                    // Handle lines with unexpected length
                    System.out.println("Data '" + line + "' is not parsed as it does not follow the schema.");
                    continue;
                }

                // Extract fields based on fixed positions
                String name = extractField(line, 1, 20).trim();
                String gender = extractField(line, 20, 21).trim();
                String age = extractField(line, 22, 25).trim();

                // Create a new Record object
                Record record = new Record(name, gender, age);
                records.add(record);
            }
        }
        return records;
    }

    private String extractField(String line, int start, int end) {
        return line.substring(start - 1, end);
    }
}

File: src/main/java/org/example/FixedLengthParser.kt
package org.example

import java.io.File
import java.io.IOException

class FixedLengthParser {
    @Throws(IOException::class)
    fun parseFile(filePath: String): List<Record> {
        val records = mutableListOf<Record>()

        File(filePath).bufferedReader().useLines { lines ->
            lines.forEach { line ->
                if (line.length == 25) {
                    // Extract fields based on fixed positions
                    val name = extractField(line, 1, 20).trim()
                    val gender = extractField(line, 20, 21).trim()
                    val age = extractField(line, 22, 25).trim()

                    // Create a new Record object
                    val record = Record(name, gender, age)
                    records.add(record)
                } else {
                    // Handle lines with unexpected length
                    println("Data '$line' is not parsed as it does not follow the schema.")
                }
            }
        }
        return records
    }

    private fun extractField(line: String, start: Int, end: Int): String {
        return line.substring(start - 1, end)
    }
}